src/box/tuple.c

```diff
@@ -796,6 +800,16 @@ box_tuple_validate(box_tuple_t *tuple, box_tuple_format_t *format)
 
 /* }}} box_tuple_* */
 
+box_tuple_t *
+box_tuple_new_with_validation(box_tuple_format_t *format, const char *data,
+			      const char *end, bool validate)
+{
+	struct tuple *ret = tuple_new(format, data, end, validate);
+	if (ret == NULL)
+		return NULL;
+	return tuple_bless(ret);
```

Comment:
```
tarantool> box.tuple.new({box.NULL}, {format = {{type = 'integer', name = 'id', is_nullable = true}}})
---
- error: 'Tuple field 1 (id) type does not match one required by operation: expected
    integer, got nil'
...
```

Should be possible since I marked field as nullable.

Reply:
The only supported format options are 'name' and 'type' (refer to design document).

Reply:
Hmm... but without nullability this feature is incomplete I believe. Probably we need to ask products about this.

Reply:
See [comment](https://github.com/tarantool/tarantool/pull/8549#discussion_r1173998353).

Reply:
Fixed, now your example will work.